ui/components/multichain/network-manager/hooks/useNetworkChangeHandlers.ts

```diff
@@ -91,6 +87,19 @@ export const useNetworkChangeHandlers = () => {
 
   const [actionMode, setActionMode] = useState(ACTION_MODE.LIST);
 
+  useEffect(() => {
+    // Fire and forget async operations for better performance
+    // setTimeout with 0 delay pushes these operations to the next event loop tick,
+    // preventing them from blocking the current execution stack and improving UI responsiveness.
+    // This technique is called "yielding to the event loop" - it allows higher priority
+    // tasks (like UI updates) to execute first before these background operations run.
+    setTimeout(() => {
+      dispatch(updateCustomNonce(''));
+      dispatch(setNextNonce(''));
+      dispatch(detectNfts(allChainIds));
```

Comment:
Discussion - do we _really_ need to detect NFTs on every network select?
Future perf - we can remove this since we have a refresh button, and when you navigate to the nft tab we fetch.